pkg/controller/controlplane/valuesprovider.go

```diff
@@ -587,6 +623,64 @@ func getCRCChartValues(
 	return values, nil
 }
 
+// getIPAMChartValues collects and returns the ipam-controller chart values.
+func getIPAMChartValues(
+	cp *extensionsv1alpha1.ControlPlane,
+	cluster *extensionscontroller.Cluster,
+	checksums map[string]string,
+	scaledDown bool,
+) (map[string]interface{}, error) {
+
+	mode := "ipv4"
+	if isIPv6(cluster) {
+		mode = "ipv6"
+	}
+	if isDualStack(cluster) {
+		mode = "dual-stack"
+	}
+
+	nodeCidrMaskSizeIPv4 := int32(24)
+	nodeCidrMaskSizeIPv6 := int32(64)
+	if cluster.Shoot.Spec.Kubernetes.KubeControllerManager != nil && cluster.Shoot.Spec.Kubernetes.KubeControllerManager.NodeCIDRMaskSize != nil {
+		if isIPv4(cluster) {
+			nodeCidrMaskSizeIPv4 = *cluster.Shoot.Spec.Kubernetes.KubeControllerManager.NodeCIDRMaskSize
+		}
+		if isIPv6(cluster) {
+			nodeCidrMaskSizeIPv6 = *cluster.Shoot.Spec.Kubernetes.KubeControllerManager.NodeCIDRMaskSize
+		}
+	}
+
+	// podNetwork is default IPv4 range for aws-ipam-controller in case IPv6 is used.
+	podNetwork := "192.168.0.0/16"
+	if slices.Contains(cluster.Shoot.Spec.Networking.IPFamilies, v1beta1.IPFamilyIPv4) {
+		podNetwork = extensionscontroller.GetPodNetwork(cluster)
```

Comment:
@DockToFuture you should rebase because with the latest vendored g/g, I believe that this method returns a slice.